### PR TITLE
HMIS 105 Report Section 1

### PR DESCRIPTION
**Pull request by moshonk**
Please find this for review @ssmusoke @carapai  
Contains HMIS105 Section 1 report with mapped Excel template

cc @ningosi

### CODE REVIEW

**Comment by ssmusoke** [comment]
@moshonk This PR has merge conflicts that need to be resolved

**Comment by moshonk** [comment]
@ssmusoke, the conflicts have been resolved
cc @carapai @ningosi

File: api/pom.xml [hunk 1]
```diff
@@ -101,6 +101,13 @@
 			<groupId>org.openmrs.module</groupId>
 			<artifactId>webservices.rest-omod-common</artifactId>
 		</dependency>
+		
+		<dependency>
+ 	         <groupId>org.openmrs.module</groupId>
+ 	         <artifactId>reportingcompatibility-api</artifactId>
+ 	         <version>2.0.3</version>
```
Comment (ssmusoke) [inline api/pom.xml]: @moshonk This dependency has to be defined in the main pom.xml not here

**Comment by ssmusoke** [comment]
@moshonk @ningosi Please coordinate on this HMIS 105 report. My understanding was that they are all in a single 105 report file - please clarify.

**Comment by ningosi** [comment]
@ssmusoke there are several templates given by @carapai  which are suppossed to cover certain sections, for that matter, it will be good to have each template with its own report file, such that when one is done with their sections and code is merged, the other one has to pull and attach their sections. I think managing them in chunks is good especially when running them, they will take short time than running all of the sections in a single file. Moreover, that is how the DHIS 2 app as designed to consume them

File: api/pom.xml (hunk 1 above)
Comment (ningosi) [inline api/pom.xml]: @moshonk I don't think we need that dependency, it is already available in the main pom

**Comment by ssmusoke** [comment]
@ningosi So is this report following the templates provided by @carapai? As I would like to stick to those which are straight from DHIS2

**Comment by ningosi** [comment]
@ssmusoke this report follow the templates provided by @carapai which will eventually be consumed by the DHIS 2 app created by METS

**Comment by ssmusoke** [comment]
@ningosi Got you - I would like to suggest that we focus on Sections 2, 4 and 5, as we have access to data to validate those reports and not Section 1 yet

File: api/pom.xml (hunk 1 above)
Comment (moshonk) [inline api/pom.xml]: Ok. I'll have it removed

**Comment by ningosi** [comment]
@moshonk you will need to refactor this PR into 2. One for section 1(OPD) and another for section 4(HCT), the way it is now, it is hard for @ssmusoke to test. He only has patient data for section 4, 2 and 5. Please prioritize that

**Comment by ssmusoke** [comment]
@moshonk Please resolve the conflicts in this PR

**Comment by ningosi** [comment]
@moshonk conflicts to ressovle

**Comment by ningosi** [comment]
This is ready for review @ssmusoke

**Comment by ssmusoke** [comment]
@moshonk new merge conflicts after the last PR merge

**Comment by ssmusoke** [comment]
@moshonk Please resolve conflicts from latest merge

**Comment by ssmusoke** [comment]
@moshonk Check the Travis failure - error stack below 

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project ugandaemrreports-api: Compilation failure: Compilation failure:
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/library/dimension/CommonReportDimensionLibrary.java:[143,42] method htcAgeGroups() is already defined in class org.openmrs.module.ugandaemrreports.reporting.library.dimension.CommonReportDimensionLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2651,28] method individualsCounselled() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2659,28] method individualsTested() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2667,28] method individualsWhoReceivedHIVTestResults() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2675,28] method individualsWhoReceivedHIVTestResultsInLast12Months() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2683,28] method individualsTestedForTheFirstTime() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2691,28] method individualsWhoTestedHivPositive() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2699,28] method individualsWhoTestedHivPositiveAndWithPresumptiveTb() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2707,28] method individualsTestedMoreThanTwiceInLast12Months() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2715,28] method individualsCounseledAndTestedAsCouple() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2723,28] method individualsTestedAndReceivedResultsAsACouple() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2731,28] method couplesWithConcordantPositiveResults() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2739,28] method couplesWithDiscordantResults() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2747,28] method individualsCounselledAndTestedForPep() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2755,28] method individualsCounselledAndTestedMarps() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2763,28] method hivPositiveIndividualsTestedAtAnEarlyStage() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary
[ERROR] /home/travis/build/METS-Programme/openmrs-module-ugandaemr-reports/api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java:[2771,28] method clientsLinkedToCare() is already defined in class org.openmrs.module.ugandaemrreports.library.Moh105IndicatorLibrary`

**Comment by ningosi** [comment]
@moshonk there seem to be a repetition of code. **Line 669 to 789** seem to have been duplicated in **Line 2651 to 2771**  if those are doing the same thing, you might consider removing one, or if they are different, you might be needed to change the methods signature

**Comment by moshonk** [comment]
@ningosi, noted, I am addressing the issue shortly

**Comment by ningosi** [comment]
Ther is a repetition of this method **htcAgeGroups()** in **CommonReportDimensionLibrary.java** also consider removing one chuck @moshonk

**Comment by moshonk** [comment]
The issues have been resolved
@ssmusoke @ningosi

**Comment by ssmusoke** [comment]
Ken I have an error that the report with the specified uuid does not exist.
I suspect that there is an error loading the report

**Comment by ningosi** [comment]
@moshonk I get this https://pastebin.com/EsSnPCCn on our demo server. It seems that report definition is not saved in the db. and also this **WARN - SerializedDefinitionServiceImpl.getDefinitionByUuid(131) |2017-08-31 13:19:11,971| Unable to deserialize Definition : Serialized org.openmrs.module.reporting.report.definition.ReportDefinition named <HMIS 105 - SECTION 1>
org.openmrs.api.db.DAOException: Unable to deserialize object: Serialized org.openmrs.module.reporting.report.definition.ReportDefinition named <HMIS 105 - SECTION 1>**

**Comment by moshonk** [comment]
@ningosi, the report definition usually fails to load if a referenced concept is missing. Since it loads properly in my dev environment, I want to set up an environment using a clean version of the ugandaemr db then test it out. cc @ssmusoke

**Comment by moshonk** [comment]
@ssmusoke, I have tested this on a clean instance of UgandaEMR and the report loads without a problem.  Can you share the database instance you are using? I'm going to try load the same in the intellisoft demo server.

**Comment by moshonk** [comment]
@ssmusoke I think I found the troublesome concept . Please check if Concept Id 130 (PUERPERAL SEPSIS) exists in your concept dictionary with the following uuid **dc66a5b3-30ab-102d-86b0-7a5022ba4115**

@ningosi, I have updated the Demo server.

**Comment by ningosi** [comment]
@ssmusoke did you see the comment from @moshonk? the report runs well here 45.79.145.240:8080/openmrs. @moshonk you need to include the DHIS 2 code as pulled form the global property

**Comment by ssmusoke** [comment]
@moshonk Found it the UUID for Puperal Sepsis is 130AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA not the other one. I remember we have a problem with this UUID in the past as it was creating a duplicate which is why the report is not loading

**Comment by ningosi** [comment]
@ssmusoke seems it reads that way on http://aijar.mets.or.ug:8080/ugandaemr/dictionary/concept.htm?conceptId=130 but different on http://45.79.145.240:8080/openmrs/dictionary/concept.htm?conceptId=130. @moshonk fix that in the code and make our demo server morror the same concept dictionary from METS

**Comment by moshonk** [comment]
@ssmusoke I have updated the code. 

The demo server is also updated @ningosi

**Pull request merged**